test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -0,0 +1,359 @@
+	 */
+	private AddressBook createDefaultAddressBook() throws Exception {
+		UniqueTagList tags = new UniqueTagList(tagDemocrat, tagRepublican);
+		UniquePersonList persons = new UniquePersonList(personBarack, personBill);
+		
+		AddressBook addressBook = new AddressBook(persons, tags);
+		
+		return addressBook;
+	}
+	
+	/**
+	 * checks if the underlying container behind an iterable is empty
+	 * 
+	 * @param it, the iterable
+	 * @return whether the container is empty
+	 */
+	private <T> boolean isEmpty(Iterable<T> it) {
+		return !it.iterator().hasNext();
+	}
+	
+	/**
+	 * checks if the underlying containers behind two iterables are equal
+	 * 
+	 * @param it0, the iterable representing the first container
+	 * @param it1, the iterable representing the second container
+	 * @return if the underlying containers are equal
+	 */
+	private <T> boolean isEqual(Iterable<T> it0, Iterable<T> it1) {
+		Iterator<T> currentPtr0 = it0.iterator();
+		Iterator<T> currentPtr1 = it1.iterator();
+		
+		while (currentPtr0.hasNext()) {
+			T val0 = currentPtr0.next();
+			T val1 = currentPtr1.next();
+			
+			if (!val0.equals(val1)) {
+				return false;
+			}
+		}
+		
+		return !currentPtr1.hasNext();
+	}
+	
+	@Before
+	public void setUp() throws Exception {
+		// Tag constructor will not throw as the strings supplied match the regex \p{Alnum}+
+		tagRepublican = new Tag("republican");
+		tagDemocrat = new Tag("democrat");
+		tagIndependent = new Tag("independent");
+		
+		// UniqueTagList constructor will not throw as there are no duplicate elements
+		personBarack = new Person(new Name("Barack"),
+                                   new Phone("155426351", false),
+                                   new Email("barack@whitehouse.gov", false),
+                                   new Address("White House", false),
+                                   new UniqueTagList(tagDemocrat));
+		personGeorge = new Person(new Name("George"),
+				                   new Phone("125469835", false),
+                                   new Email("george@whitehouse.gov", false),
+                                   new Address("White House", false),
+                                   new UniqueTagList(tagRepublican));
+		personBill   = new Person(new Name("Bill"),
+                                   new Phone("154982351", false),
+                                   new Email("bill@whitehouse.gov", false),
+                                   new Address("White House", false),
+                                   new UniqueTagList(tagDemocrat));
+		personMonroe = new Person(new Name("Monroe"),
+                                   new Phone("168463354", false),
+                                   new Email("email@notinvented.yet", false),
+                                   new Address("James Monroe Tomb, Richmond, Virginia", false),
+                                   new UniqueTagList(tagIndependent, tagDead));
```

Comment:
When defining test data, try to follow a pattern that is easy to remember. For example, the first person's name has initials A. B. (e.g. Alice Betsy), second person's name has initials B. C. (Ben Charles) and so on. When things always follow a pattern, anything going out of that pattern (e.g. Ben suddenly appears at the start of the list) gets noticed immediately and sometimes can lead to discovering bugs.